### ISSUE TITLE
Compile error with wrong magic word after replacing older version.

### ISSUE THREAD

**Issue by MoreporkGithub** [ruffle-rs/ruffle#4606]
**Describe the bug**
<!--
If you have a question about Ruffle, you can ask for help on our Discord chat:
https://discord.gg/ruffle

No, I can't. Discord won't let me login even though I am a member and have done so before. It says something about a different location, but I am using the same email address...

**Here's my problem...**
I have spent a number of hours on the following problem, and I have read all the docs I can find on the site.

I want to replace and test the latest version of Ruffle on a localhost web site running IIS.

It should be a simple matter of downloading the right zip to the "ruffle" directory on the localhost web site, extracting the files, and the existing  .js stuff on the affected pages should work. as it did before. I did that, but it simply doesn't work.

I can't find an easy way to "uninstall" Ruffle so I can ensure a clean new install... In the end I deleted the Ruffle directory and re-created it.

There is no problem with the WASM type because it was set up for the previous version and worked. There is no problem with the .swf file because, if I download the Windows 32 version of ruffle.exe, it displays it correctly. It is the current webhosting version that does not work. It used to have a link to make a bug report, but now it refers to text. I duly read this and it was no help at all. It says to make sure the "required files" are available, but it doesn't tell me what the "required files" ARE, or how it expects them to be available. The ONLY files I have are the ones in the downloaded .zip and they have been extracted and are available in a directory under the Web Site root, which is referenced as explained in the Ruffle docs with a path on the pages that need to use Ruffle.

Here is the detail report from Ruffle:

# Error Info
Error name: CompileError
Error message: WebAssembly.instantiate(): expected magic word 00 61 73 6d, found 3c 21 44 4f @+0
Error stack:
```
CompileError: WebAssembly.instantiate(): expected magic word 00 61 73 6d, found 3c 21 44 4f @+0
```

# Player Info
Allows script access: true
Player type: Object
SWF URL: images/quill1.swf
Attribute 0: undefined
Attribute 1: undefined
Attribute 2: undefined

# Page Info
Page URL: http://localhost/PRIMAMetro/Scripting.aspx

# Browser Info
Useragent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.101 Safari/537.36 Edg/91.0.864.48
OS: Win32

# Ruffle Info
Version: 0.1.0
Name: nightly 2021-06-17
Channel: nightly
Built: 2021-06-17T00:11:56.220Z
Commit: 622f35eb713c224a980ab67eb251a4b6880e1834

Also consult the FAQ for common issues and questions:
https://github.com/ruffle-rs/ruffle/wiki/Frequently-Asked-Questions-For-Users
I did this but I couldn't see this issue there.

Please avoid submitting reports on these known issues:
 * ActionScript 3.0 is not supported (#1368)
 * Filter effects (drop shadow, glow, blur, etc.) are not supported (#15)
 * Blend modes are not supported (#58)
 * Strokes are rendered too thin/thick (#1955)

**Expected behavior**

I expect it to render the SWF image as it used to in the past.

**Is the problem with the Ruffle desktop app, extension, or self-hosted version?**
I believe it is the self-hosted version.

**What platform are you using?**
 - If you're using the desktop version, please list the OS you are using.
 - If you're using the web version, please list the OS, browser, and mobile device (if applicable).
 - 
 -The web version. OS is Windows 10, web server is IIS version 7, Browser is Edge. All Windows updates have been applied.

I'd really like to get the Ruffle functionality back so if anyone can assist, I'll be grateful.

Thanks,

Pete.

**Comment by EmperorBale** [comment]
Did you replace every file? This usually happens when you either didn't replace the .js file or the .wasm file. This can also happen if your server is improperly caching your files

**Comment by EmperorBale** [comment]
Could you also post the error message so we can see the exact issue

**Comment by MoreporkGithub** [comment]
@EmperorBale 

Thanks for responding. I am having major problems with simply posting the problem. There should be a button here that allows people unfamiliar with how these responses are edited by the system to simply start over. I've spent 10 minutes just trying to post the problem.

Anyway, here's as much as I can salvage:

I have spent a number of hours on the following problem, and I have read all the docs I can find on the site.

I want to replace and test the latest version of Ruffle on a localhost web site running IIS.

It should be a simple matter of downloading the right zip to the "ruffle" directory on the localhost web site, extracting the files, and the existing  .js stuff on the affected pages should work. as it did before. I did that, but it simply doesn't work.

I can't find an easy way to "uninstall" Ruffle so I can ensure a clean new install... In the end I deleted the Ruffle directory and re-created it.

There is no problem with the WASM type because it was set up for the previous version and worked. There is no problem with the .swf file because, if I download the Windows 32 version of ruffle.exe, it displays it correctly. It is the current webhosting version that does not work. It used to have a link to make a bug report, but now it refers to text. I duly read this and it was no help at all. It says to make sure the "required files" are available, but it doesn't tell me what the "required files" ARE, or how it expects them to be available. The ONLY files I have are the ones in the downloaded .zip and they have been extracted and are available in a directory under the Web Site root, which is referenced as explained in the Ruffle docs with a path on the pages that need to use Ruffle.

Here is the detail report from Ruffle:

# Error Info
Error name: CompileError
Error message: WebAssembly.instantiate(): expected magic word 00 61 73 6d, found 3c 21 44 4f @+0
Error stack:
```
CompileError: WebAssembly.instantiate(): expected magic word 00 61 73 6d, found 3c 21 44 4f @+0
```

# Player Info
Allows script access: true
Player type: Object
SWF URL: images/quill1.swf
Attribute 0: undefined
Attribute 1: undefined
Attribute 2: undefined

# Page Info
Page URL: http://localhost/PRIMAMetro/Scripting.aspx

# Browser Info
Useragent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.101 Safari/537.36 Edg/91.0.864.48
OS: Win32

# Ruffle Info
Version: 0.1.0
Name: nightly 2021-06-17
Channel: nightly
Built: 2021-06-17T00:11:56.220Z
Commit: 622f35eb713c224a980ab67eb251a4b6880e1834

Also consult the FAQ for common issues and questions:
https://github.com/ruffle-rs/ruffle/wiki/Frequently-Asked-Questions-For-Users
I did this but I couldn't see this issue there.

Please avoid submitting reports on these known issues:
 * ActionScript 3.0 is not supported (#1368)
 * Filter effects (drop shadow, glow, blur, etc.) are not supported (#15)
 * Blend modes are not supported (#58)
 * Strokes are rendered too thin/thick (#1955)

**Expected behavior**
A clear and concise description of what you expected to happen.

I expect it to render the SWF image as it used to in the past.

**Is the problem with the Ruffle desktop app, extension, or self-hosted version?**
I believe it is the self-hosted version.

**What platform are you using?**
 - If you're using the desktop version, please list the OS you are using.
 - If you're using the web version, please list the OS, browser, and mobile device (if applicable).
 - 
 -The web version. OS is Windows 10, web server is IIS version 7, Browser is Edge. All Windows updates have been applied.

I'd really like to get the Ruffle functionality back so if anyone can assist, I'll be grateful.

Thanks,

Pete.

**Comment by MoreporkGithub** [comment]
@EmperorBale 

To answer your question: I deleted the entire directory and replaced it empty, before adding the new zip file and extracting it.

There could not be any old files as far as I know.

**Comment by EmperorBale** [comment]
Looks like your wasm file is being served in ASCII bytes, which means your wasm file is either corrupted or maybe the MIME type is wrong. It's much more likely your MIME type is wrong. Have you confirmed your .wasm files are being served with the `application/wasm` MIME type? If you don't know what that is, I can send you more information on that to help you out

**Comment by MoreporkGithub** [comment]
The WASM file is exactly as extracted from the Ruffle zip. I think it is probably the MIME type, as you say. I didn't confirm it because I forgot what file these things are in... I think it is .htaccess but I believe that is only for Apache servers. Anyway, I went into IIS and set the MIME type for WASM. It wasn't there, so somehow it must have been removed or maybe IIS was re-installed and I forgot...

I retried and everything worked as it should!

Many thanks for your help and clear comments
